### PR TITLE
Do not regenerate folder ids when for v2.x to v1 (ignore `retainIds`)

### PR DESCRIPTION
**Pull request by kamalaknn**

### CODE REVIEW

File: test/unit/v2.0.0/converter-v2-to-v1.test.js [hunk 1]
```diff
@@ -711,6 +712,48 @@ describe('v2.0.0 to v1.0.0', function () {
...
+
+                folderOrderIds = folderOrderIds.concat(result.folders_order);
+
+                _.forEach(result.folders, function (folder) { // eslint-disable-line lodash/prefer-map
+                    folderIds.push(folder.id);
+                    folderOrderIds = folderOrderIds.concat(folder.folders_order);
+                    requestOrderIds = requestOrderIds.concat(folder.order);
+                });
+
+                _.forEach(result.requests, function (request) { // eslint-disable-line lodash/prefer-map
```
Comment (kunagpal) [inline test/unit/v2.0.0/converter-v2-to-v1.test.js]: `requestIds = _.map(result.request, 'id');`

File: lib/converters/v2.0.0/converter-v2-to-v1.js [hunk 2]
```diff
@@ -548,9 +552,13 @@ _.assign(Builders.prototype, {
...
                 variables = self.variables(folder),
-                id = folder.id || folder._postman_id,
+
+                // do not attempt to regenerate folder id here when `retainIds` is set to false
+                // if id is changed here the parent's `folder_order` also needs to be changed
+                // that can't be done yet
+                id = folder.id || folder._postman_id || util.uid(),
                 result = {
-                    id: self.options.retainIds && id ? id : util.uid(),
+                    id: id,
```
Comment (kunagpal) [inline lib/converters/v2.0.0/converter-v2-to-v1.js]: Can be assigned directly.

File: test/unit/v2.0.0/converter-v2-to-v1.test.js [hunk 3]
```diff
@@ -711,6 +712,48 @@ describe('v2.0.0 to v1.0.0', function () {
...
+                    requestIds.push(request.id);
+                });
+
+                expect(folderIds.length).to.be.ok;
```
Comment (kunagpal) [inline test/unit/v2.0.0/converter-v2-to-v1.test.js]: `expect(folderIds).to.be.an.instanceOf(Array).that.is.not.empty;`

File: test/unit/v2.0.0/converter-v2-to-v1.test.js [hunk 4]
```diff
@@ -711,6 +712,48 @@ describe('v2.0.0 to v1.0.0', function () {
...
+                expect(requestIds.length).to.be.ok;
+
+                // validate the format of request and folder ids
+                _.forEach(folderIds, function (folderId) {
```
Comment (kunagpal) [inline test/unit/v2.0.0/converter-v2-to-v1.test.js]: Can be replaced with `_.every`

**Comment by kunagpal** [comment]
@kamalaknn As discussed, normalization and v1 -> v2.x flows **remain affected**, will have to look at them later. The proper fix is to introduce a parent reference for folders, requests, and responses and transform children first, followed by a parent reference update.

**Comment by kamalaknn** [comment]
@kunagpal Yes, we should be working fixing parent to child references next. That will allow us to properly fix `retainIds:false` across convertors and normalization.

**Pull request merged**